util/conditions/patch_test.go

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+)
+
+func TestGetPatch(t *testing.T) {
+	fooTrue := TrueCondition("foo")
+	fooFalse := FalseCondition("foo", "reason foo", clusterv1.ConditionSeverityInfo, "message foo")
+
+	tests := []struct {
+		name    string
+		current clusterv1.Conditions
+		target  clusterv1.Conditions
+		want    []ConditionPatch
+	}{
+		{
+			name:    "No changes return empty patch",
+			current: conditionList(),
+			target:  conditionList(),
+			want:    nil,
+		},
+		{
+			name:    "No changes return empty patch",
+			current: conditionList(fooTrue),
+			target:  conditionList(fooTrue),
+			want:    nil,
+		},
+		{
+			name:    "Detects AddConditionPatch",
+			current: conditionList(),
+			target:  conditionList(fooTrue),
+			want: []ConditionPatch{
+				{
+					Base:   nil,
+					Target: fooTrue,
+					Op:     AddConditionPatch,
+				},
+			},
+		},
+		{
+			name:    "Detects ChangeConditionPatch",
+			current: conditionList(fooTrue),
+			target:  conditionList(fooFalse),
+			want: []ConditionPatch{
+				{
+					Base:   fooTrue,
+					Target: fooFalse,
+					Op:     ChangeConditionPatch,
+				},
+			},
+		},
+		{
+			name:    "Detects ChangeConditionPatch",
```

Comment:
Should be `Detects RemoveConditionPatch`

Reply:
fixed